### PR TITLE
Playground: Set background image

### PR DESCRIPTION
**Pull request by sanchitmalhotra126**
Handles the SET_BACKGROUND_IMAGE message and sets the background image of the playground based on the file name.

I also added a generic file load error callback that can be used if there is an error loading an asset file. Javabuilder will also be verifying that the file exists in the student's project, but this provides some fallback validation. The error displays like:

Screen Shot 2021-09-22 at 12 23 32 PM
## Links
JIRA: https://codedotorg.atlassian.net/browse/CSA-843
Spec: https://docs.google.com/document/d/1Moo2s5EXZRp5rMg1VW9jlOqs_GeMN5yjU8FJgoqOEMk/edit#heading=h.yth48qvin8gl

## Testing story
Testing: local (all the things level) & unit

## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: apps/src/javalab/Playground.js [hunk 1]
```diff
@@ -1,14 +1,39 @@
...
+    assetsApi
+  ) {
     this.onOutputMessage = onOutputMessage;
     this.onNewlineMessage = onNewlineMessage;
     this.onJavabuilderMessage = onJavabuilderMessage;
     this.isGameRunning = false;
     this.isGameOver = false;
+    this.levelName = levelName;
+    this.starterAssetFilenames = [];
+    this.starterAssetsApi = starterAssetsApi || starterAssets;
```
Comment (bencodeorg) [inline apps/src/javalab/Playground.js:24]: Are these two assignments just to support tests? Maybe add comment if so?

File: apps/src/javalab/Playground.js [hunk 2]
```diff
@@ -103,4 +132,18 @@ export default class Playground {
       return;
     }
   }
+
+  getUrl(filename) {
```
Comment (bencodeorg) [inline apps/src/javalab/Playground.js:151]: Do we assume Javabuilder has already done the work of confirming that the file exists?

File: apps/src/javalab/PlaygroundVisualizationColumn.jsx [hunk 3]
```diff
@@ -60,6 +64,15 @@ const styles = {
     width: 800,
     height: 800,
     overflow: 'hidden'
+  },
+  playgroundBackground: {
+    position: 'absolute',
+    width: 800,
```
Comment (bencodeorg) [inline apps/src/javalab/PlaygroundVisualizationColumn.jsx:70]: What's the distinction between the 400x400 size we use in Javabuilder vs. 800x800 here? Do we just scale between the two?

File: apps/src/javalab/Playground.js [hunk 4]
```diff
@@ -89,6 +114,10 @@ export default class Playground {
       // can't set background if game is over
       return;
     }
+
+    const backgroundElement = this.getBackgroundElement();
```
Comment (molly-moen) [inline apps/src/javalab/Playground.js:129]: should we reset the background to have no image and 0 opacity in `reset`?

**Review by molly-moen — COMMENTED** [review]
One question about reset, otherwise looks good!

File: apps/src/javalab/PlaygroundVisualizationColumn.jsx [hunk 5]
```diff
@@ -40,6 +40,10 @@ class PlaygroundVisualizationColumn extends React.Component {
         <div style={{opacity}}>
           <ProtectedVisualizationDiv>
             <div id="playground-container" style={styles.playground}>
+              <img
```
Comment (bencodeorg) [inline apps/src/javalab/PlaygroundVisualizationColumn.jsx:43]: No issues with having no src initially?

**Review by bencodeorg — APPROVED** [review]
LGTM!

File: apps/src/javalab/Playground.js (hunk 4 above)
Comment (sanchitmalhotra126) [reply apps/src/javalab/Playground.js:129]: Ah yes good call, I'll add that

File: apps/src/javalab/Playground.js (hunk 1 above)
Comment (sanchitmalhotra126) [reply apps/src/javalab/Playground.js:24]: Yep they're just for testing - I'll add a comment

File: apps/src/javalab/Playground.js (hunk 2 above)
Comment (sanchitmalhotra126) [reply apps/src/javalab/Playground.js:151]: So eventually, yes, Javabuilder would have confirmed that this file exists in the student's project. We could also do some client-side validation though - I'll add an onerror callback we can use

File: apps/src/javalab/PlaygroundVisualizationColumn.jsx (hunk 3 above)
Comment (sanchitmalhotra126) [reply apps/src/javalab/PlaygroundVisualizationColumn.jsx:70]: Yeah, so I believe 800x800 is the maximum size the visualization can have and then we scale it based on available space. The 400x400 represents the pixel coordinate system that students use for setting x, y, width, height. Those coordinates and sizes are scaled to the actual size of the visualization. That scaling logic is here: https://github.com/code-dot-org/code-dot-org/blob/8e789eb2c579aba273fc4c5ef7bd75148791f7bd/apps/src/javalab/JavalabPanels.jsx#L149

File: apps/src/javalab/PlaygroundVisualizationColumn.jsx (hunk 5 above)
Comment (sanchitmalhotra126) [reply apps/src/javalab/PlaygroundVisualizationColumn.jsx:43]: Shouldn't be since we're setting the opacity to 0

**Review by molly-moen — APPROVED** [review]
LGTM!

**Pull request merged**